### ISSUE TITLE
ignore_errors parameter cannot ignore the specified exceptions.

### ISSUE THREAD

**Issue by yuqiuwen** [getsentry/sentry-python#2677]
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

```python
sentry_sdk.init(
        dsn=os.getenv("SENTRY_DSN"),
        integrations=[FlaskIntegration()],
        environment=env,
        ignore_errors=[AuthException]
)
```

`AuthException` inherits from Exception

### Expected Result

ignore `AuthException`

### Actual Result

`AuthException` is still captured

### Product Area

Unknown

**Comment by sentrivana** [comment]
Hey @yuqiuwen, thanks for writing in.

I can't reproduce the issue. Here's my small Flask app:

```python
import sentry_sdk
from flask import Flask

class AuthException(Exception):
    pass

sentry_sdk.init(
    dsn=<DSN>,
    debug=True,
    ignore_errors=[AuthException],
)

app = Flask(__name__)

@app.route("/autherror")
def auth_error():
    raise AuthException('oh no')
    return "ok"
```

Calling `/autherror` results in an Internal Server Error, as expected, but the `AuthException` is not sent to Sentry.

Could you provide a minimal reproducible example to help us reproduce?

**Comment by yuqiuwen** [comment]
> Hey @yuqiuwen, thanks for writing in.
> 
> I can't reproduce the issue. Here's my small Flask app:
> 
> ```python
> import sentry_sdk
> from flask import Flask
> 
> class AuthException(Exception):
>     pass
> 
> sentry_sdk.init(
>     dsn=<DSN>,
>     debug=True,
>     ignore_errors=[AuthException],
> )
> 
> app = Flask(__name__)
> 
> @app.route("/autherror")
> def auth_error():
>     raise AuthException('oh no')
>     return "ok"
> ```
> 
> Requesting `/autherror` results in an Internal Server Error, as expected, but the `AuthException` is not sent to Sentry.
> 
> Could you provide a minimal reproducible example to help us reproduce?

this is my code:
```python
class AuthException(Exception):
    def __init__(self, message="", code: Enum = AppCode.AUTH_ERROR, errmsg=None):
        super().__init__(message)
        self.code = code.value
        self.errmsg = errmsg
```

however, it worked for me using `before_send`
```python
def before_send(event, hint):
    if "exc_info" in hint:
        exc_type, exc_value, tb = hint["exc_info"]
        if isinstance(exc_value, (AuthException,)):
            return None
        return event
```

**Comment by sentrivana** [comment]
@yuqiuwen `before_send` is another way to do it, great that that works for you. I'd still like to understand why `ignore_errors` doesn't though, but for that I'd need a full example of a small app where `ignore_errors` doesn't work. So far with my small app as outlined above (with your `AuthException`) it all works as expected.

**Comment by yuqiuwen** [comment]
> @yuqiuwen `before_send` is another way to do it, great that that works for you. I'd still like to understand why `ignore_errors` doesn't though, but for that I'd need a full example of a small app where `ignore_errors` doesn't work. So far with my small app as outlined above (with your `AuthException`) it all works as expected.

Maybe I see , because of logging the error, is right?
if i don't log the error, it works well.
```python
@app.errorhandler(AuthException)
def handle_auth_error(e):
    logger.error(traceback.format_exc())
    return make_response(str(e), code=e.code, errmsg=e.errmsg)
```

**Comment by sentrivana** [comment]
> Maybe I see , because of logging the error, is right? if i don't log the error, it works well.
> 
> ```python
> @app.errorhandler(AuthException)
> def handle_auth_error(e):
>     logger.error(traceback.format_exc())
>     return make_response(str(e), code=e.code, errmsg=e.errmsg)
> ```

Right, it's our logging integration capturing it. Thanks for digging into it!

I think this could be solved by improving our [ignored error detection](https://github.com/getsentry/sentry-python/blob/e864eab559c2b37b44bdf6f353cbdb25c8f885ce/sentry_sdk/client.py#L430) to do something similar like you did in your `before_send`.

I will put this in our backlog.

**Comment by yuqiuwen** [comment]
> > Maybe I see , because of logging the error, is right? if i don't log the error, it works well.
> > ```python
> > @app.errorhandler(AuthException)
> > def handle_auth_error(e):
> >     logger.error(traceback.format_exc())
> >     return make_response(str(e), code=e.code, errmsg=e.errmsg)
> > ```
> 
> Right, it's our logging integration capturing it. Thanks for digging into it!
> 
> I think this could be solved by improving our [ignored error detection](https://github.com/getsentry/sentry-python/blob/e864eab559c2b37b44bdf6f353cbdb25c8f885ce/sentry_sdk/client.py#L430) to do something similar like you did in your `before_send`.
> 
> I will put this in our backlog.

here's another question. Why is the log output incomplete?
![image](https://github.com/getsentry/sentry-python/assets/12853176/d97798af-7354-4b88-8344-3fdd91035ceb)

**Comment by yuqiuwen** [comment]
> > here's another question. Why is the log output incomplete?
> 
> @yuqiuwen Where are you seeing this? Can you please send the URL of the page in Sentry?

This is the output of error.log, the part after the red line is truncated in the sentry
![image](https://github.com/getsentry/sentry-python/assets/12853176/09b7971f-37bd-4cdd-8d04-47c250e5fd0e)

**Comment by sentrivana** [comment]
It probably got truncated by some part of the system because it was too long. Not sure off the top of my head if this is the SDK or the server doing this. Potentially might also just be a display thing. If you check the event JSON in Sentry (the button with the curly brackets), do you see the whole message?

**Comment by yuqiuwen** [comment]
> It probably got truncated by some part of the system because it was too long. Not sure off the top of my head if this is the SDK or the server doing this. Potentially might also just be a display thing. If you check the event JSON in Sentry (the button with the curly brackets), do you see the whole message?

 click the button still only shows part of the log
![image](https://github.com/getsentry/sentry-python/assets/12853176/c936132a-115e-4e9f-8986-d5dc650500c8)

**Comment by sl0thentr0py** [comment]
@yuqiuwen that is just our server side truncation because the message is too long. 
The ideal way is to not let this captured by logging but as a normal exception so that this goes into the stacktrace and not serialized into the message string.

You can also use `logging.error('bla', exc_info=True)` and the stacktrace will be added to the sentry event.

**Comment by szokeasaurusrex** [comment]
Closing, I think everything is resolved here. Please reopen if you still need assistance
